mbuild/formats/lammpsdata.py

```diff
@@ -777,17 +927,30 @@ def write_lammpsdata(
                 atom_line = "{index:d}\t{zero:d}\t{type_index:d}\t{charge:.4e}\t{x:.6f}\t{y:.6f}\t{z:.6f}\n"
 
         for i, coords in enumerate(xyz):
-            data.write(
-                atom_line.format(
-                    index=i + 1,
-                    type_index=unique_types.index(types[i]) + 1,
-                    zero=structure.atoms[i].residue.idx + moleculeID_offset,
-                    charge=charges[i],
-                    x=coords[0],
-                    y=coords[1],
-                    z=coords[2],
+            if unit_style == "lj":
+                data.write(
+                    atom_line.format(
+                        index=i + 1,
+                        type_index=unique_types.index(types[i]) + 1,
+                        zero=structure.atoms[i].residue.idx + moleculeID_offset,
+                        charge=charges[i],
+                        x=coords[0] / NM_TO_ANG,
```

Comment:
Same comments as above with the following [division](https://github.com/mosdef-hub/mbuild/blob/f12c4c5db103d3ebe557277b885df2a3bb3a6ad8/mbuild/formats/lammpsdata.py#L284)

Reply:
Same comments as above. In this case, we do the conversion before writing coordinates IFF the user has selected reduced units instead of updating the coords array. The conversion is only needed for reduced units so this fix does not impact cases with real units.